SFE_* targets don't compile with blinky-baremetal

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect

<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->

SFE_* targets fails to compile blinky-baremetal in Mbed OS 6

Arm and GCC Error

```
[mbed] Working path "/Users/.../Mbed Programs/mbed-os-example-blinky-baremetal" (program)
Building project mbed-os-example-blinky-baremetal (SFE_ARTEMIS, GCC_ARM)
Scan: mbed-os-example-blinky-baremetal
Configuration error: Bootloader not supported on this target. ROM start not found in targets.json.
Configuration error: Bootloader not supported on this target. RAM start not found in targets.json.
Compile [  5.8%]: hci_drv_apollo3.c
[Fatal Error] hci_drv_apollo3.c@52,10: wsf_types.h: No such file or directory
[ERROR] ./mbed-os/connectivity/drivers/ble/FEATURE_BLE/TARGET_Ambiq_Micro/hal/apollo3/hci_drv_apollo3.c:52:10: fatal error: wsf_types.h: No such file or directory
   52 | #include "wsf_types.h"
      |          ^~~~~~~~~~~~~
compilation terminated.
```

#### Target(s) affected by this defect ?


SFE_ARTEMIS
SFE_ARTEMIS_ATP
SFE_ARTEMIS_DK
SFE_ARTEMIS_MODULE
SFE_ARTEMIS_NANO
SFE_ARTEMIS_THING_PLUS
SFE_EDGE
SFE_EDGE2



#### Toolchain(s) (name and version) displaying this defect ?

Arm Compiler 6 and GCC 9

#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in /platform/mbed_version.h. The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
 
    Master branch is indicated by 'mbed-os-99.99.99
    
    For an issue found on Master please provide the sha being used.
-->

mbed-os-6.4.0

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)

Mbed CLI 1.10.4

#### How is this defect reproduced ? 

```
mbed import https://github.com/ARMmbed/mbed-os-example-blinky-baremetal
cd mbed-os-example-blinky
mbed compile -t <toolchain> -m <target>
```

@ARMmbed/team-sparkfun

--- a/connectivity/drivers/ble/FEATURE_BLE/TARGET_Ambiq_Micro/mbed_lib.json
+++ b/connectivity/drivers/ble/FEATURE_BLE/TARGET_Ambiq_Micro/mbed_lib.json
@@ -1,0 +1,3 @@
+{
+    "name": "cordio-ambiq-micro"
+}

